hazelcast/src/main/java/com/hazelcast/spi/impl/operationexecutor/classic/ClassicOperationExecutor.java

```diff
@@ -65,58 +70,44 @@
     private final ILogger logger;
 
     // all operations for specific partitions will be executed on these threads, e.g. map.put(key, value)
-    private final PartitionOperationThread[] partitionOperationThreads;
+    private final PartitionOperationThread[] partitionThreads;
     private final OperationRunner[] partitionOperationRunners;
 
-    private final OperationQueue genericOperationQueue;
+    private final OperationQueue genericQueue = new DefaultOperationQueue();
 
     // all operations that are not specific for a partition will be executed here, e.g. heartbeat or map.size()
-    private final GenericOperationThread[] genericOperationThreads;
+    private final GenericOperationThread[] genericThreads;
     private final OperationRunner[] genericOperationRunners;
 
     private final Address thisAddress;
-    private final NodeExtension nodeExtension;
-    private final HazelcastThreadGroup threadGroup;
     private final OperationRunner adHocOperationRunner;
-    private final MetricsRegistry metricsRegistry;
 
     public ClassicOperationExecutor(GroupProperties properties,
                                     LoggingService loggerService,
                                     Address thisAddress,
                                     OperationRunnerFactory operationRunnerFactory,
-                                    HazelcastThreadGroup hazelcastThreadGroup,
+                                    HazelcastThreadGroup threadGroup,
                                     NodeExtension nodeExtension,
                                     MetricsRegistry metricsRegistry) {
         this.thisAddress = thisAddress;
-        this.nodeExtension = nodeExtension;
-        this.threadGroup = hazelcastThreadGroup;
-        this.metricsRegistry = metricsRegistry;
         this.logger = loggerService.getLogger(ClassicOperationExecutor.class);
-        this.genericOperationQueue = new DefaultOperationQueue();
 
         this.adHocOperationRunner = operationRunnerFactory.createAdHocRunner();
 
         this.partitionOperationRunners = initPartitionOperationRunners(properties, operationRunnerFactory);
-        this.partitionOperationThreads = initPartitionThreads(properties);
+        this.partitionThreads = initPartitionThreads(properties, threadGroup, nodeExtension, metricsRegistry);
 
         this.genericOperationRunners = initGenericOperationRunners(properties, operationRunnerFactory);
-        this.genericOperationThreads = initGenericThreads();
+        this.genericThreads = initGenericThreads(threadGroup, nodeExtension, metricsRegistry);
 
-        logger.info("Starting with " + genericOperationThreads.length + " generic operation threads and "
-                + partitionOperationThreads.length + " partition operation threads.");
-    }
+        metricsRegistry.scanAndRegister(this, "operation");
```

Comment:
Instructions can be re-ordered? This is a huge no-go in our favorite book from Brian Goetz (3.2.1 Safe construction practices). So it maybe wrong in other parts of the system as well.